src/analyze/analyze-verify.c

```diff
@@ -215,6 +215,9 @@ static int verify_unit(Unit *u, bool check_man) {
         if (k < 0 && r == 0)
                 r = k;
 
+        if (found_error_when_parsing_config && r == 0)
+                r = -EINVAL;
```

Comment:
This will return an error in perpetuity after returning an error once. This function should return an error only when *new* errors happen while it is executing. Please turn  `found_error_when_parsing_config`  into a counter, and make `log_syntax` simply increase the variable if appropriate, and here record the value at entry and then return an error only if the value grew in the meantime. This is nicer because it doesn't assume that there's just one user of this.

Reply:
Do you think having the global variable just reset to false again after r is assigned -EINVAL will be appropriate here? That could address the newest issue about the perpetual error too and would do away with the auxiliary counter variable. 
So something like:
if (log_syntax_issue_reported && r == 0) {
                r = -EINVAL;
                log_syntax_issue_reported = false;
}

Reply:
It would, but the whole thing would still be fairly useless, because … see my reply to @bluca's comment below.